components/menu/sub_menu.tsx

```diff
@@ -0,0 +1,239 @@
+        event.preventDefault();
+        setAnchorElement(null);
+    }
+
+    // This handleKeyDown is on the menu item which opens the submenu
+    function handleSubMenuParentItemKeyDown(event: KeyboardEvent<HTMLLIElement>) {
+        if (event.key === 'ArrowRight' || event.key === 'Enter') {
+            event.preventDefault();
+            setAnchorElement(event.currentTarget);
+        }
+    }
+
+    function handleSubMenuKeyDown(event: KeyboardEvent<HTMLUListElement>) {
+        if (event.key === 'ArrowDown' || event.key === 'ArrowUp') {
+            // Stop the event from propagating upwards since that causes navigation to move by 2 items at a time
+            event.stopPropagation();
+        } else if (event.key === 'ArrowLeft' || event.key === 'Escape') {
+            event.preventDefault();
+            setAnchorElement(null);
+        }
+    }
+
+    useEffect(() => {
+        if (anyModalOpen && !isMobileView) {
+            setAnchorElement(null);
+        }
+    }, [anyModalOpen, isMobileView]);
+
+    const originOfAnchorAndTransform = useMemo(() => getOriginOfAnchorAndTransform(forceOpenOnLeft, anchorElement), [anchorElement]);
+
+    const hasSubmenuItems = Boolean(children);
+    if (!hasSubmenuItems) {
+        return null;
+    }
+
+    const passedInTriggerButtonProps = {
+        id,
+        'aria-controls': menuId,
+        'aria-haspopup': true,
+        'aria-expanded': isSubMenuOpen,
+        disableRipple: true,
+        leadingElement,
+        labels,
+        trailingElements,
+        isDestructive,
+        onClick: handleSubMenuOpen,
+    };
+
+    if (isMobileView) {
+        return (<ParentMenuItem {...passedInTriggerButtonProps}/>);
+    }
+
+    return (
+        <ParentMenuItem
+            {...rest} // pass through other props which might be coming in from the material-ui
+            {...passedInTriggerButtonProps}
+            onMouseEnter={handleSubMenuOpen}
+            onMouseLeave={handleSubMenuClose}
+            onKeyDown={handleSubMenuParentItemKeyDown}
+        >
+            <MuiMenuStyled
+                anchorEl={anchorElement}
+                open={isSubMenuOpen}
+                asSubMenu={true}
+                anchorOrigin={originOfAnchorAndTransform.anchorOrigin}
+                transformOrigin={originOfAnchorAndTransform.transformOrigin}
+                sx={{pointerEvents: 'none'}} // disables the menu background wrapper for accessing submenu
+            >
+                <MuiMenuList
+                    id={menuId}
+                    component='ul'
+                    aria-label={menuAriaLabel}
+                    className={A11yClassNames.POPUP}
+                    onKeyDown={handleSubMenuKeyDown}
+                    sx={{
+                        pointerEvents: 'auto', // reset pointer events to default from here on
+                        paddingTop: 0,
+                        paddingBottom: 0,
+                    }}
```

Comment:
nit: its better to define those values beforehand to reduce unnecessary rerenders